experimental/language-generation/docs/lg-file-format.md

```diff
@@ -120,36 +167,49 @@ Hello, good morning
+## Entities 
+- When used directly within a one-of variation text, entity references are denoted by enclosing them in curly brackets - @{`entityName`}
+- When used as a parameter within a 
+    - [pre-built function][4] or 
+    - as a parameter to [template resolution call](#Parametrization-of-templates) or 
+    - a condition in [conditional response template](#conditional-response-template)
+they are simply expressed as `entityName`. - e.g. @{entityName == null}
+## Using pre-built functions in variations
+[Pre-built functions][4] supported by the [Common expression language][3] can also be used inline in a one-of variation text to achieve even more powerful text composition. To use an expression inline, simply wrap it in curly brackets - @{}.
 
-### Parametrization of templates
-To aid with contextual re-usability, templates can be parametrized. With this different callers to the template can pass in different values for use in expansion resolution.
-
-Here is an example of a template parametrization. 
+Here is an example that illustrates that - 
 
 ```markdown
-# timeOfDayGreetingTemplate (param1)
-- IF: @{param1 == 'morning'}
-    - good morning
-- ELSEIF: @{param1 == 'afternoon'}
-    - good afternoon
-- ELSE: 
-    - good evening
+# RecentTasks
+- IF: @{count(recentTasks) == 1}
+    - Your most recent task is @{recentTasks[0]}. You can let me know if you want to add or complete a task.
+- ELSEIF: @{count(recentTasks) == 2}
+    - Your most recent tasks are @{join(recentTasks, ', ', ' and ')}. You can let me know if you want to add or complete a task.
+- ELSEIF: @{count(recentTasks) > 2}
+    - Your most recent @{count(recentTasks)} tasks are @{join(recentTasks, ', ', ' and ')}. You can let me know if you want to add or complete a task.
+- ELSE:
+    - You don't have any tasks.
+```
+The above example uses the [join][5] pre-built function to list all values in the `recentTasks` collection. 
 
-# morningGreeting
-- @{timeOfDayGreetingTemplate('morning')}
+If template name is the same with builtin function's name, template will be executed first, without automatically executing one according to the parameter variable. 
 
-# timeOfDayGreeting
-- @{timeOfDayGreetingTemplate(timeOfDay)}
-```
+In case where you must have the template name be the same as a prebuilt function name, you can use `builtin.xxx` to refer to the prebuilt function and avoid possible collisions with your own template name.
 
-## Entities 
-- When used directly within a one-of variation text, entity references are denoted by enclosing them in curly brackets - {`entityName`}
-- When used as a parameter within a 
-    - [pre-built function][4] or 
-    - as a parameter to [template resolution call](#Parametrization-of-templates) or 
-    - a condition in [conditional response template](#conditional-response-template)
-they are simply expressed as `entityName`.
+Here is an example that illustrates that
+
+```markdown
+# length(a)
+- This is use's customized length function
```

Comment:
param is 'a' is not used. 
Is it mandatory to have a param? if optional, use it? or mentioned something about it

Reply:
Fixed.